wdl-format/src/task.rs

```diff
@@ -0,0 +1,428 @@
+//! A module for formatting elements in tasks.
+
+use wdl_ast::v1::CommandPart;
+use wdl_ast::v1::CommandSection;
+use wdl_ast::v1::Decl;
+use wdl_ast::v1::RequirementsItem;
+use wdl_ast::v1::RequirementsSection;
+use wdl_ast::v1::RuntimeItem;
+use wdl_ast::v1::RuntimeSection;
+use wdl_ast::v1::TaskDefinition;
+use wdl_ast::v1::TaskItem;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+
+use super::comments::format_inline_comment;
+use super::comments::format_preceding_comments;
+use super::first_child_of_kind;
+use super::state::SPACE;
+use super::Formattable;
+use super::State;
+use super::NEWLINE;
+
+impl Formattable for CommandSection {
+    fn format<T: std::fmt::Write>(&self, writer: &mut T, state: &mut State) -> std::fmt::Result {
+        format_preceding_comments(
+            &SyntaxElement::from(self.syntax().clone()),
+            writer,
+            state,
+            false,
+        )?;
+
+        let command_keyword = first_child_of_kind(self.syntax(), SyntaxKind::CommandKeyword);
+        state.indent(writer)?;
+        write!(writer, "{}", command_keyword)?;
+        format_inline_comment(&command_keyword, writer, state, true)?;
+
+        if self.is_heredoc() {
+            let open_heredoc = first_child_of_kind(self.syntax(), SyntaxKind::OpenHeredoc);
+            format_preceding_comments(&open_heredoc, writer, state, true)?;
+            // Open braces should ignore the "+1 rule" followed by other interrupted
+            // elements.
+            if state.interrupted() {
+                state.reset_interrupted();
+                state.indent(writer)?;
+            } else {
+                write!(writer, "{}", SPACE)?;
+            }
+            write!(writer, "{}", open_heredoc)?;
+        } else {
+            let open_brace = first_child_of_kind(self.syntax(), SyntaxKind::OpenBrace);
+            format_preceding_comments(&open_brace, writer, state, true)?;
+            // Open braces should ignore the "+1 rule" followed by other interrupted
+            // elements.
+            if state.interrupted() {
+                state.reset_interrupted();
+                state.indent(writer)?;
+            } else {
+                write!(writer, "{}", SPACE)?;
+            }
+            write!(writer, "{}", open_brace)?;
+        }
+
+        for part in self.parts() {
+            match part {
+                CommandPart::Text(t) => {
+                    write!(writer, "{}", t.as_str())?;
+                }
+                CommandPart::Placeholder(p) => {
+                    write!(writer, "{}", p.syntax())?;
```

Comment:
Is this writing the placeholder verbatim? Should we be formatting its expression?

Reply:
Noted in the PR description, `Expr` is a barrel of worms I'm working on in another branch. So yes, this PR just copies placeholders and expressions verbatim. This PR is already quite long, and `Expr` formatting is going to add a decent amount to the line count.

Reply:
Oh whoops, I went strait to the files tab and missed the description! Agreed that we can format these in follow-up PRs.